aws/lambda/opensearch-gha-jobs/test_lambda_function.py

```diff
@@ -0,0 +1,311 @@
+import json
+from unittest import TestCase
+from unittest.mock import Mock
+
+from lambda_function import (
+    extract_dynamodb_key,
+    extract_dynamodb_table,
+    remove_document,
+    to_number,
+    unmarshal,
+    upsert_document,
+)
+
+
+def test_extract_dynamodb_table():
+    cases = [
+        {
+            "arn": "",
+            "expected": None,
+            "description": "Invalid input - empty input",
+        },
+        {
+            "arn": "FOOBAR",
+            "expected": None,
+            "description": "Invalid input - not in ARN format",
+        },
+        {
+            "arn": "arn:aws:dynamodb:us-east-1:12345:table/torchci-workflow-job/stream/2022-01-14T01:31:51.775",
+            "expected": "torchci-workflow-job",
+            "description": "An event coming from DynamoDB",
+        },
+    ]
+
+    for case in cases:
+        arn = case["arn"]
+        expected = case["expected"]
+        TestCase().assertEqual(expected, extract_dynamodb_table({"eventSourceARN": arn}))
```

Comment:
One more nit: Could you add some description here to help a future debugger understand what went wrong (perhaps without even having to read the test code)? 

Something like: 

```suggestion
        TestCase().assertEqual(expected, extract_dynamodb_table({"eventSourceARN": arn}), "Received unexpected dynamo table name")
```